sources/packages/api/src/services/disbursement-schedule-service/disbursement-schedule-service.ts

```diff
@@ -305,4 +329,127 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
+   * @param locationId
+   * @param disbursementScheduleId
+   * @returns Disbursement and Application Summary.
+   */
+  async getDisbursementAndApplicationSummary(
+    locationId: number,
+    disbursementScheduleId: number,
+  ): Promise<DisbursementSchedule> {
+    return this.repo
+      .createQueryBuilder("disbursementSchedule")
+      .select([
+        "disbursementSchedule.id",
+        "disbursementSchedule.disbursementDate",
+        "application.id",
+        "application.applicationStatus",
+        "application.assessmentWorkflowId",
+      ])
+      .innerJoin("disbursementSchedule.application", "application")
+      .innerJoin("application.location", "location")
+      .where("location.id = :locationId", { locationId })
+      .andWhere("disbursementSchedule.id = :disbursementScheduleId", {
+        disbursementScheduleId,
+      })
+      .andWhere("application.applicationStatus IN (:...status)", {
+        status: [ApplicationStatus.enrollment, ApplicationStatus.completed],
+      })
+      .getOne();
+  }
+
+  /**
+   * Deny COE for an application.
+   ** Note: If an application has 2 COEs, and if the first COE is Rejected then 2nd COE is implicitly rejected.
+   * @param disbursementScheduleIds
+   * @param userId User who denies the COE.
+   */
+  async updateCOEToDeny(
+    applicationId: number,
+    userId: number,
+    coeDeniedReasonId: number,
+    otherReasonDesc: string,
+  ): Promise<void> {
+    this.repo
+      .createQueryBuilder()
+      .update(DisbursementSchedule)
+      .set({
+        coeStatus: COEStatus.declined,
+        coeUpdatedBy: { id: userId },
+        coeUpdatedAt: new Date(),
+        coeDeniedReason: { id: coeDeniedReasonId },
+        coeDeniedOtherDesc:
+          coeDeniedReasonId === COE_DENIED_REASON_OTHER_ID
+            ? otherReasonDesc
+            : null,
+      })
+      .where("application.id = :applicationId", { applicationId })
+      .andWhere("coeStatus = :required", { required: COEStatus.required })
+      .execute();
+  }
+
+  /**
+   * Returns the oldest/first COE which is waiting for confirmation.
+   ** This information is required to validate if the user is approving the first COE before second.
+   * @param applicationId
+   * @returns Disbursement
+   */
+  async getFirstOutstandingCOE(
+    applicationId: number,
+  ): Promise<DisbursementSchedule> {
+    return this.repo
+      .createQueryBuilder("disbursementSchedule")
+      .select(["disbursementSchedule.id"])
+      .innerJoin("disbursementSchedule.application", "application")
+      .where("application.id = :applicationId", { applicationId })
+      .andWhere("application.applicationStatus IN (:...status)", {
+        status: [ApplicationStatus.enrollment, ApplicationStatus.completed],
+      })
+      .andWhere("disbursementSchedule.coeStatus = :required", {
+        required: COEStatus.required,
+      })
+      .orderBy("disbursementSchedule.disbursementDate")
```

Comment:
The code can limit the results to only one adding .limit(1), what is the expectation of max records to be retrieved.

Reply:
limit added.